### PR TITLE
Fix Symfony IriConverter with `item_uri_template`

### PR DESCRIPTION
**Pull request by vincentchalamon**
| Q             | A
| ------------- | ---
| Branch?       | 3.1
| Tickets       | #5662
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

File: features/hydra/item_uri_template.feature [hunk 1]
```diff
@@ -131,8 +131,7 @@ Feature: Exposing a collection of objects should use the specified operation to
     """
 
   Scenario: Get a collection referencing another resource for its IRI
-    When I add "Content-Type" header equal to "application/json"
-    And I send a "GET" request to "/item_referenced_in_collection"
```
Comment (soyuka) [inline features/hydra/item_uri_template.feature:135]: please revert this change

File: src/JsonLd/Serializer/ItemNormalizer.php [hunk 2]
```diff
@@ -105,6 +105,9 @@ public function normalize(mixed $object, string $format = null, array $context =
 
         $context['api_normalize'] = true;
 
+        /* @see https://github.com/api-platform/core/pull/5663 */
+        unset($context['item_uri_template']);
```
Comment (soyuka) [inline src/JsonLd/Serializer/ItemNormalizer.php:109]: :+1:

File: features/hydra/item_uri_template.feature (hunk 1 above)
Comment (soyuka) [reply features/hydra/item_uri_template.feature:135]: sorry I thought that we wanted to use a json "Accept" on this request, indeed it looks like its a mistake

**Review by soyuka — COMMENTED** [review]
Should we remove that last parameter if you removed it alltogether?

```
    private function generateSymfonyRoute(object|string $resource, int $referenceType = UrlGeneratorInterface::ABS_PATH, Operation $operation = null, array $context = [], **Operation $identifiersExtractorOperation = null**): string
```

IIRC I did this because of huge performance improvements on item IRI generation, should we try to profile this to check for a change in performance on these code paths?

**Comment by soyuka** [comment]
https://github.com/vincentchalamon/core/pull/2/commits/72a2c715db9ae756e4a321ecb53b52b0139e64aa 

I did some cleanup and improved the patch a bit can you look at my commit (was based on your branch, rebased on 3.1)

**Comment by vincentchalamon** [comment]
> [vincentchalamon@72a2c71](https://github.com/vincentchalamon/core/commit/72a2c715db9ae756e4a321ecb53b52b0139e64aa)
> 
> I did some cleanup and improved the patch a bit can you look at my commit (was based on your branch, rebased on 3.1)
> 
> /edit: I need to check the CI

I need to investigate your commit on api-platform/demo to ensure it still fixes my original bug.

**Comment by vincentchalamon** [comment]
@soyuka this new version of the code does not work on api-platform/demo (cf. https://github.com/api-platform/demo/pull/327)
